pkg/deployment/os.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+
+	dataplaneutil "github.com/openstack-k8s-operators/dataplane-operator/pkg/util"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	ansibleeev1alpha1 "github.com/openstack-k8s-operators/openstack-ansibleee-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InstallOS ensures the node Operating System is installed
+func InstallOS(ctx context.Context, helper *helper.Helper, obj client.Object, sshKeySecret string, inventoryConfigMap string) error {
+
+	tasks := []dataplaneutil.Task{
+		{
+			Name:          "Install edpm_bootstrap",
+			RoleName:      "edpm_bootstrap",
+			RoleTasksFrom: "bootstrap.yml",
+			Tags:          []string{"edpm_bootstrap"},
+		},
+		{
+			Name:     "Install edpm_kernel",
+			RoleName: "edpm_kernel",
+			Tags:     []string{"edpm_kernel"},
+		},
+		{
+			Name:          "Install edpm_podman",
+			RoleName:      "edpm_podman",
+			RoleTasksFrom: "install.yml",
+			Tags:          []string{"edpm_podman"},
+		},
+		{
+			Name:          "Install edpm_sshd",
+			RoleName:      "edpm_sshd",
+			RoleTasksFrom: "install.yml",
+			Tags:          []string{"edpm_sshd"},
+		},
+		{
+			Name:          "Install chrony",
+			RoleName:      "chrony",
+			RoleTasksFrom: "install.yml",
+			Tags:          []string{"chrony"},
+		},
+		{
+			Name:          "Install edpm_ovn",
+			RoleName:      "edpm_ovn",
+			RoleTasksFrom: "install.yml",
+			Tags:          []string{"edpm_ovn"},
+		},
+	}
+	role := ansibleeev1alpha1.Role{
+		Name:           "Deploy EDPM Operating System Install",
+		Hosts:          "{{ primary_role_name | default([]) }}:{{ deploy_target_host | default('overcloud') }}",
```

Comment:
this should just be "all". These other group names are carry overs from TripleO that we don't need

Reply:
done!